win32/wsyslog.c

```diff
@@ -97,7 +97,7 @@ void vsyslog(int priority, const char *message, va_list args)
 
 	/* default event source */
 	if (INVALID_HANDLE_VALUE == PW32G(log_source))
-		openlog("php", LOG_PID, LOG_SYSLOG);
```

Comment:
should this be really changed with use of `PG`. I think this is a win re-implementation of syslog so it should be probably internal (it's kind of what glibc implements and we don't change glibc... :))

Reply:
Sorry, not following.  This is really the essence of the fix: allowing configuring `syslog.facility` and `syslog.ident` and having these values be used with `openlog()`.

Reply:
@bukka Anyone with basic php-src karma can do changes to win32/, its only Zend/ and TSRM/ that has an extra karma level. Both me and @weltling should be able to help out if there is an issue with any Windows stuff

Reply:
php_syslog calls openlog...

Reply:
But the `WIN32` instance of `php_syslog()` doesn't check to see if `openlog()` has been called...

Reply:
Or are you saying we should call `php_openlog()` here as well?

Reply:
Ah ok sorry I completely missed that it has got it's own instance. Not really sure why it has to be different but I guess it's probably ok.

I would still like to check with @weltling or @KalleZ if it's appropriate to use `PG` in `win32` as it seems to me like an independent part. Alternatively it could be moved to the WIN32 instance in `php_syslog.c`

Reply:
If we add the same check in the WIN32 version of `php_syslog()` then we know that `log_source` will be valid (it gets set in WIN32 `openlog()`) and we can remove:

```
	/* default event source */
 	if (INVALID_HANDLE_VALUE == PW32G(log_source))
 		openlog("php", LOG_PID, LOG_SYSLOG);
```

Reply:
Okay, pushing new changes.